modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C

```diff
@@ -839,11 +842,22 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::computeStressFinalize(
     const Real negative_wall_dislocations = MetaPhysicL::raw_value(_wall_dislocations[_qp]);
     _cell_dislocations[_qp] = _old_input_values[_cell_output_index];
     _wall_dislocations[_qp] = _old_input_values[_wall_output_index];
-    mooseException("The negative values of the cell dislocation density, ",
-                   negative_cell_dislocations,
-                   ", and/or wall dislocation density, ",
-                   negative_wall_dislocations,
-                   ". Cutting timestep.");
+    if (_window_failure[_cell_input_index] == WindowFailure::ERROR ||
+        _window_failure[_wall_input_index] == WindowFailure::ERROR)
+    {
+      mooseError("The negative values of the cell dislocation density, ",
```

Comment:
This is new, do you think it's important to error here, or is exception good enough? This is not checking input windows, its just checking if its negative. The error I am pushing for is important for the _inputs_ going into the ROM, the _outputs_ here are more about being sane. `WindowFailure` does not really describe the output value checks does it?.

Reply:
You're right, this should only be an exception and not an error.